### PR TITLE
Change naming for Notification Resources

### PR DESCRIPTION
**Pull request by kcherukupalli**
## What’s changing?

Change naming for "Notification" Resources to "Notify".   Updated IAC scripts, source code and documentation.

## Why?

nac-1617
nac-1618
nac-1619

## This PR has:

- [x] **JIRA ticket(s)** (_Must be listed in PR request and branch name should follow convention NAC-123-description_)<description>

- [ ] **Security Related Changes** (_Security Tab questions have been answered on JIRA ticket(s)_)

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Documented API changes** (_required on public classes and methods, and elsewhere as appropriate_)
    - [ ] External API documentation - OpenApi docs and Generated Docs (e.g. https://github.com/18F/piipan/tree/dev/docs/openapi) 
    - [ ] Internal API documentation (e.g. https://github.com/18F/piipan/tree/dev/metrics/docs/openapi)   

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [x] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)
    - [x] Have IaC scripts run successfully against an existing deployed environment?
    - [x] Have IaC scripts run successfully against a newly deployed environment?
    - [ ] Do the IaC changes require deleting existing resources?

## Anything else?

_Add other details that are helpful for review here_

## Code Review Checklist:

- [ ] Automated checks all pass? (code builds successfully, tests pass, test coverage meets/exceeds threshold, etc.) 
- [ ] Have all requirements and Acceptance Criteria been met?
- **If there are code changes**
    - [ ] Is there no unnecessary complexity? Code is readable, maintainable, etc.
    - [ ] Do the changes follow SOLID principles?
    - [ ] Are the changes robust? Concurrency considered? Performance? Security (e.g. no sql injections)?
    - [ ] Are newly added dependencies needed and appropriate? Are dependency files (e.g. packages.json & packages.lock.json) updated?
    - [ ] For any new projects/solutions, have they been added to code coverage files?
    - [ ] Public classes & methods are documented    
- **If there are API changes**
    - [ ] Does API make sense? Useful and concise?
    - [ ] Are existing API interfaces/contracts intact? i.e. no breaking changes
    - [ ] Are API tests included?
    - [ ] Are API responses appropriate and documented? (200s, 201, 404s, etc.)
    - [ ] Are internals kept private and not leaked into the API?
    - [ ] External API documentation - OpenApi docs and Generated Docs 
    - [ ] Internal API documentation    
- **If there are IAC changes**
    - [ ] IAC Scripts are readable & maintainable
    - [ ] IAC Scripts are documented
    - [ ] IAC scripts pass required checks
    - [ ] IAC scripts were run successfully by author
- **General Logging Questions**
    - [ ] Do new capabilities (or changed capabilities) include logging?
    - [ ] Is PII not logged?
- **General Error Handling Questions**
    - [ ] Has exception handling been added for new capabilities (or changed capabilities)?
    - [ ] Are errors/exceptions properly communicated to end-users? Were there requirements for user facing error messages?
    - [ ] Are errors/exceptions logged?
    - [ ] Is PII kept out of error/exception logging?
- **If there are Developer Documentation Changes**
    - [ ] General Project documentation (under piipan\docs. e.g. iac.md, architecture.md, etc.)
    - [ ] Github documentation additions/changes (Readmes)
- **General automated test questions**
    - [ ] End-to-End Tests provided (if code implements new user story)
    - [ ] Are tests sufficient?
    - [ ] Corner cases considered?
    - [ ] Are performance tests needed?

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -116,7 +116,7 @@ _Target date: TBD_
 
 **Acceptance criteria**
 - At least 3 states are using the NAC for every required case action
-- The NAC sends email notifications to each state involved in a match when a match is found
+- The NAC sends email notification to each state involved in a match when a match is found
```
Comment (bgfoshay) [inline docs/roadmap.md:119]: Question/Nitpick: This may have been an accidental change? I think it should still be plural

File: match/src/Piipan.Match/Piipan.Match.Core/Services/MatchResNotifyService.cs [hunk 2]
```diff
@@ -4,39 +4,39 @@
...
-using Piipan.Notifications.Core.Services;
+using Piipan.Notify.Common.Models;
+using Piipan.Notify.Core.Services;
 using Piipan.States.Core.DataAccessObjects;
 
 namespace Piipan.Match.Core.Services
 {
     /// <summary>
-    /// This service will send out notifications for Match Res Events, if they meet criteria for sending notifications.
+    /// This service will send out notify for Match Res Events, if they meet criteria for sending notify.
```
Comment (bgfoshay) [inline match/src/Piipan.Match/Piipan.Match.Core/Services/MatchResNotifyService.cs:14]: Nitpick: you can probably leave this documenation at "notifications" so that the documentation sounds correct

File: match/src/Piipan.Match/Piipan.Match.Core/Services/MatchResNotifyService.cs [hunk 3]
```diff
@@ -4,39 +4,39 @@
...
+            INotifyService notifyService)
         {
             _stateInfoDao = stateInfoDao;
-            _notificationService = notificationService;
+            _notifyService = notifyService;
         }
 
         /// <summary>
-        /// Sends a Match notification out if the updates warrant a notification.
+        /// Sends a Match notify out if the updates warrant a notify.
```
Comment (bgfoshay) [inline match/src/Piipan.Match/Piipan.Match.Core/Services/MatchResNotifyService.cs:32]: Nitpick: same nitpick regarding documentation grammar

File: match/src/Piipan.Match/Piipan.Match.Core/Services/MatchResNotifyService.cs [hunk 4]
```diff
@@ -4,39 +4,39 @@
...
         /// </summary>
         /// <param name="matchResRecordBeforeUpdate">The match before the update</param>
         /// <param name="matchResRecordAfterUpdate">The match after the update</param>
         /// <returns></returns>
-        public async Task SendNotification(MatchDetailsDto matchResRecordBeforeUpdate, MatchDetailsDto matchResRecordAfterUpdate)
+        public async Task SendNotify(MatchDetailsDto matchResRecordBeforeUpdate, MatchDetailsDto matchResRecordAfterUpdate)
```
Comment (bgfoshay) [inline match/src/Piipan.Match/Piipan.Match.Core/Services/MatchResNotifyService.cs:37]: Nitpick: Would this sound better if it was just "Notify" rather than "SendNotify"?

**Comment by kcherukupalli** [comment]
As per conversation opening new PR #4386 with changed AC.
